### ISSUE TITLE
[WebProfilerBundle] Redirects are not detected

### ISSUE THREAD

**Issue by vudaltsov** [symfony/symfony#24730]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Symfony version  | 3.4

```php
<?php

namespace App\Controller;

use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
use Symfony\Component\HttpFoundation\Response;

class IndexController extends AbstractController
{
    /**
     * @Route(name="index")
     */
    public function indexAction()
    {
        return $this->redirectToRoute('redirect');
    }

    /**
     * @Route("/redirect", name="redirect")
     */
    public function redirectAction()
    {
        return new Response('<html><body></body></html>');
    }
}
```

![image](https://user-images.githubusercontent.com/2552865/32139114-6d946dce-bc49-11e7-8f96-41344cebb900.png)

**Comment by kejwmen** [comment]
Is not related to [configuration](https://symfony.com/doc/current/reference/configuration/web_profiler.html#intercept-redirects)?

**Comment by vudaltsov** [comment]
@kejwmen, no. Profiler should show `Redirected from ...` regardless of `web_profiler.intercept_redirects`.

**Comment by sroze** [comment]
This feature would only work if you have the session enabled as you can see in [the `RequestDataCollector`](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpKernel/DataCollector/RequestDataCollector.php#L134). Can you confirm you have the same issue with sessions enabled?

**Comment by sroze** [comment]
Maybe you don't have the session started (anymore?) since @nicolas-grekas' updates. Not sure how we can have this feature working without session tbh...

**Comment by vudaltsov** [comment]
The session was not enabled in the `framework.session` config...

With
```yaml
framework:
    session:
        handler_id: ~
```
Everything works as expected on 3.4 beta-2.

My fault. But with flex you have to me much more attentive not to forget to enable things.
I was testing other stuff so far and didn't think of the session at all.

**Comment by weaverryan** [comment]
Yea... this worries me a little bit actually... are there any other spots where not having the session enabled might silently cause some weird issues?

**Comment by sroze** [comment]
@weaverryan same. I've proposed this PR to fix the issue: #24774
